bob-common/src/data.rs

```diff
@@ -193,6 +194,24 @@ impl BobOptions {
         }
     }
 
+    pub fn new_delete(options: Option<DeleteOptions>) -> Self {
```

Comment:
It is time to split `BobOptions` into `BobPutOptions`, `BobGetOptions` and `BobDeleteOptions`